### PR TITLE
ON-25448 # Added `autoDenyAfterClarificationRequestDays` to approvalC…

### PR DESCRIPTION
**Pull request by Zaxist**
…onfiguration on form

### CODE REVIEW

File: typescript/forms.d.ts [hunk 1]
```diff
@@ -840,6 +840,12 @@ export type Form = {
      * when denying an approval
      */
     denyCannedResponses?: FormApprovalCannedResponse[]
+    /**
+     * The amount of days after a clarification request has been sent with no
+     * response until the approval is automatically denied. Set `undefined`
+     * or unset for no Auto Deny.
+     */
+    autoDenyAfterClarificationRequestDays?: number
```
Comment (kizaonline) [inline typescript/forms.d.ts:848]: We need fields for this stuff as well 

> User can set the number of days after clarification to deny an approval on a per form basis, private notes, notify parties, canned response and notification notes

File: typescript/forms.d.ts [hunk 2]
```diff
@@ -840,6 +840,26 @@ export type Form = {
...
+     * Set `undefined` or unset for no Auto Deny.
+     */
+    autoDenyAfterClarificationRequest?: {
+      /**
+       * The amount of days after a clarification request has been sent with no
+       * response until the approval is automatically denied.
+       */
+      days: number
+      /** The email addresses of the users to be notified of the result */
+      notificationEmailAddress?: string[]
```
Comment (kizaonline) [inline typescript/forms.d.ts:855]: This should a boolean to indicate if they want to send a notification (the address will be the default address if configured/available)

Comment (kizaonline) [reply typescript/forms.d.ts:855]: Perhaps we shoud keep this field as well and add the default address to this later when sending?

Comment (Zaxist) [reply typescript/forms.d.ts:855]: Hmm, feels like it should have the same options as the approvals front end

Comment (kizaonline) [reply typescript/forms.d.ts:855]: Cool, we'll keep it then and add another field for choosing to notify

**Review by kizaonline — APPROVED** [review]

**Pull request merged**